src/bindings/c/src/ov_infer_request.cpp

```diff
@@ -295,7 +295,7 @@ ov_status_e ov_infer_request_wait_for(ov_infer_request_t* infer_request, const i
     }
     CATCH_OV_EXCEPTIONS
 
-    return ret ? ov_status_e::OK : ov_status_e::UNEXPECTED;
+    return ret ? ov_status_e::OK : ov_status_e::RESULT_NOT_READY;
```

Comment:
I suppose the change should be ported to 2023.3

@ilya-lavrenov

Reply:
agree